### PR TITLE
Publisher Coverage Script and Badge

### PR DESCRIPTION
**Pull request by dobbersc**
This PR adds a script to execute a baseline real-time crawler validation. A problem with the current crawler unit tests is that the crawlers are only validated on pre-fetched articles. This script aims to check a publisher's availability and basic crawl results.
The tests include a real-time crawl for each publisher's news map and RSS Feed checking the received articles for attribute completeness. Note that this script does not check the attributes' correctness, only their presence. The resulting publisher coverage is then summarised in the output report.

To automate the publisher coverage script this PR includes an action that runs every day at 01:00 generating a badge containing the publisher coverage displayed in the README. See [here](https://github.com/flairNLP/fundus/actions/runs/6632110282/job/18017082728) for a current action run.

To enhance the functionality we could also run the tests for each source independently, e.g. news map and RSS feed if specified, such that a crawler only passes the tests if all specified sources return appropriate articles.

### CODE REVIEW

**Comment by MaxDall** [comment]
@dobbersc Thanks for adding this, that's exactly what we needed.

**Review by MaxDall — APPROVED** [review]

File: scripts/publisher_coverage.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    )
+
+    for publisher_region in publisher_regions:
+        print(f"{publisher_region.__name__:-^50}")
+
+        publisher: PublisherEnum
+        for publisher in sorted(
+            publisher_region, key=lambda p: cast(PublisherEnum, p).name  # type: ignore[no-any-return]
+        ):
+            publisher_name: str = publisher.name  # type: ignore[attr-defined]
```
[the remark this answers is not in the document]
Comment (dobbersc) [reply scripts/publisher_coverage.py:31]: I totally agree with you. On my local machine, this part is also not necessary. Although, for some reason, the CI fails without this change ([CI Log](https://github.com/flairNLP/fundus/actions/runs/6632372288/job/18017916604)). Just to be sure I'll try it again without. The enum typing is unfortunately a bit messy :/.

[the remark this answers is not in the document]
Comment (dobbersc) [reply scripts/publisher_coverage.py]: This is very weird. I'm running the same version of mypy on a complete fresh conda installation of fundus (cleared cache etc.) and get no complaints locally from mypy while the CI's mypy fails.

[the remark this answers is not in the document]
Comment (MaxDall) [reply scripts/publisher_coverage.py]: Thanks for testing this. That is indeed very weird.

> The enum typing is unfortunately a bit messy :/

I agree. I will add something about this to #325 

Do you have any clue what's causing this discrepancy between CI and local runs? My next suggestion would be to remove the `mypy` version constraints from `pyproject.toml` and see if this is a version-specific problem or some problem with the CI in general. What do you think?

**Review by MaxDall — APPROVED** [review]

**Pull request merged**